syntax = "proto3";
package mappers;

// Server queries
service Mapper {
	rpc ProcessFiles(ProcessFilesRequest) returns (ProcessFilesResponse) {}
	rpc ReturnKV(reducerID) returns (KV) {}
}
message ProcessFilesRequest {
	repeated string filenames = 1;
  string task = 2;
  int32 num_reducers = 3;
}
message ProcessFilesResponse {
	bool response = 1;
}
message reducerID {
	int32 id = 1;
}
message KV {
	string dict_object = 1;
}